bundler/lib/bundler/definition.rb

```diff
@@ -517,7 +517,29 @@ def materialize(dependencies)
     end
 
     def precompute_source_requirements_for_indirect_dependencies?
-      @remote && sources.non_global_rubygems_sources.all?(&:dependency_api_available?) && !sources.aggregate_global_source?
+      precompute = false
+      if @remote
+        if sources.non_global_rubygems_sources.all?(&:dependency_api_available?)
+          precompute = true
+        else
+          non_depeneency_api_warning
+        end
+      end
```

Comment:
I think maybe the implementation is simpler like this:

```ruby
def precompute_source_requirements_for_indirect_dependencies?
  return false unless @remote && !sources.aggregate_global_source?

  if sources.non_global_rubygems_sources.all?(&:dependency_api_available?)
    true
  else
    non_dependency_api_warning
    false
  end
end
````

Reply:
OK. It's simpler. I will update the logic with your suggestion.

Note previously the `sources.non_global_rubygems_sources.all?(&:dependency_api_available?)` is evaluated earlier than `!sources.aggregate_global_source?`. Now in the new logic, the `!sources.aggregate_global_source?` is evaluated earlier than `sources.non_global_rubygems_sources.all?(&:dependency_api_available?)`.

Reply:
Indeed. I don't think these methods have side effects so in principle it shouldn't make a difference.